### PR TITLE
[feature] logstash pulsar input support token

### PR DESCRIPTION
**Pull request by JunFu0814**
logstash pulsar input support token

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,6 +51,25 @@ input{
...
+    serviceUrl => "pulsar://127.0.0.1:6650"
+    codec => "plain"
+    topics => [ 
+        "persistent://public/default/topic1", 
+        "persistent://public/default/topic2"
+    ]
+    subscriptionName => "my_subscription"
+    subscriptionType => "Shared"
+    subscriptionInitialPosition => "Earliest"
+    enable_token => true
```
Comment (nodece) [inline README.md:67]: Rename to `enable_auth`?

File: README.md [hunk 2]
```diff
@@ -6,25 +6,36 @@ It is fully free and fully open source. The license is Apache 2.0, meaning you a
...
+| serviceUrl                       |                            string                            |       Yes |
+| topics                           |                            array                             |       Yes |
+| subscriptionName                 |                            string                            |       Yes |
+| consumerName                     |                            string                            |       Yes |
+| subscriptionType                 | string, one of["Shared","Exclusive","Failover","Key_shared"] |        No |
+| subscriptionInitialPosition      |             string, one of["Latest","Earliest"]              |        No |
+| codec                            |               string, one of ["plain","json"]                |        No |
+| protocols                        |           array, ciphers list. default is TLSv1.2            |        No |
+| ciphers                          |                      array,ciphers list                      |        No |
+| enable_tls                       |       boolean, one of [true, false]. default is false        |        No |
```
Comment (nodece) [inline README.md:27]:
```suggestion

```

We can check the scheme of the `serviceUrl` to determine whether TLS is enabled.

Example:
- https://pulsar.app
- pulsar+ssl://pulsar.app:6650

File: README.md [hunk 3]
```diff
@@ -6,25 +6,36 @@ It is fully free and fully open source. The license is Apache 2.0, meaning you a
...
+| tls_trust_store_path             |        string, required if enable_tls is set to true         |        No |
```
Comment (nodece) [inline README.md:28]: This is used for Java KeyStore, we should also support the PEM file, which is more general.

File: README.md [hunk 4]
```diff
@@ -6,25 +6,36 @@ It is fully free and fully open source. The license is Apache 2.0, meaning you a
...
+| tls_trust_store_password         |                   string, default is empty                   |        No |
+| enable_tls_hostname_verification |       boolean, one of [true, false]. default is false        |        No |
+| allow_tls_insecure_connection    |        boolean, one of [true, false].default is false        |        No |
+| enable_token                     |        boolean, one of [true, false].default is false        |        No |
```
Comment (nodece) [inline README.md:32]: Rename to `enable_auth`?

File: README.md [hunk 5]
```diff
@@ -6,25 +6,36 @@ It is fully free and fully open source. The license is Apache 2.0, meaning you a
...
+| auth_plugin_class_name           |                            string                            |        No |
```
Comment (nodece) [inline README.md:33]: Rename to `auth-plugin`?

File: README.md [hunk 6]
```diff
@@ -6,25 +6,36 @@ It is fully free and fully open source. The license is Apache 2.0, meaning you a
...
+| auth_plugin_params_string        |                            string                            |        No |
```
Comment (nodece) [inline README.md:34]: Rename to `auth-params`?

File: README.md (hunk 4 above)
Comment (JunFu0814) [reply README.md:32]: Maybe rename it to `enable_jwt` more make sense

File: README.md (hunk 6 above)
Comment (JunFu0814) [reply README.md:34]: Rename to `auth_params`?

File: README.md (hunk 5 above)
Comment (JunFu0814) [reply README.md:33]: Rename to `auth_plugin`?

File: README.md (hunk 4 above)
Comment (nodece) [reply README.md:32]: This is used for multiple authentication plugins, so as the token, oauth2, tls, and basic, so on.

File: README.md (hunk 6 above)
Comment (nodece) [reply README.md:34]: Right.

Comment (nodece) [reply README.md:34]: Same with the pulsar-admin params.

File: src/main/java/org/apache/pulsar/logstash/inputs/Pulsar.java [hunk 7]
```diff
@@ -105,8 +105,14 @@ public class Pulsar implements Input {
     private static final PluginConfigSpec<String> CONFIG_TLS_TRUST_STORE_PASSWORD =
             PluginConfigSpec.stringSetting("tls_trust_store_password","");
 
-    private static final PluginConfigSpec<String> CONFIG_AUTH_PLUGIN_CLASS_NAME =
-            PluginConfigSpec.stringSetting("auth_plugin_class_name",authPluginClassName);
+    private static final PluginConfigSpec<Boolean> CONFIG_ENABLE_AUTH =
+            PluginConfigSpec.booleanSetting("enable_auth",false);
+
+    private static final PluginConfigSpec<String> CONFIG_AUTH_PLUGIN =
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/logstash/inputs/Pulsar.java:111]: I'm sorry for making an incorrect review. This change breaks the` auth_plugin_class_name` config. Please revert this.

File: src/main/java/org/apache/pulsar/logstash/inputs/Pulsar.java [hunk 8]
```diff
@@ -157,7 +168,13 @@ private void createConsumer() throws PulsarClientException {
                         .enableTlsHostnameVerification(enableTlsHostnameVerification)
                         .tlsTrustStorePath(tlsTrustStorePath)
                         .tlsTrustStorePassword(config.get(CONFIG_TLS_TRUST_STORE_PASSWORD))
-                        .authentication(config.get(CONFIG_AUTH_PLUGIN_CLASS_NAME),authMap)
+                        .authentication(config.get(CONFIG_AUTH_PLUGIN),authMap)
```
Comment (nodece) [inline src/main/java/org/apache/pulsar/logstash/inputs/Pulsar.java:171]:
```suggestion
                        .authentication(authPluginClassName, authMap)
```

**Review by nodece — DISMISSED** [review]

**Review by nodece — DISMISSED** [review]

File: src/main/java/org/apache/pulsar/logstash/inputs/Pulsar.java [hunk 9]
```diff
@@ -322,7 +324,12 @@ public Collection<PluginConfigSpec<?>> configSchema() {
                 CONFIG_SUBSCRIPTION_TYPE,
                 CONFIG_SUBSCRIPTION_INITIAL_POSITION,
                 CONFIG_CONSUMER_NAME,
-                CONFIG_CODEC
+                CONFIG_CODEC,
+                CONFIG_AUTH_PLUGIN,
+                // Pulsar Token Config
+                CONFIG_AUTH_PARAMS
+
```
Comment (Shoothzj) [inline src/main/java/org/apache/pulsar/logstash/inputs/Pulsar.java:331]: unnecessary blank lines?
Can we add a tailing comma?

**Comment by JunFu0814** [comment]
@Shoothzj please review again ,thx

**Review by nodece — APPROVED** [review]

**Review by nodece — APPROVED** [review]

**Pull request merged**